### PR TITLE
[ServiceBus&EventHubs] remove RetryMode enum

### PR DESCRIPTION
**Pull request by swathipil**
follow-up to #21884 and #22317. 
After API Review: use core `RetryMode` instead of EH/SB RetryMode

### CODE REVIEW

File: sdk/eventhub/azure-eventhub/azure/eventhub/_configuration.py [hunk 1]
```diff
@@ -10,15 +10,15 @@
...
+from azure.core.pipeline.policies import RetryMode
 
 
 class Configuration(object):  # pylint:disable=too-many-instance-attributes
     def __init__(self, **kwargs):
         self.user_agent = kwargs.get("user_agent")  # type: Optional[str]
         self.retry_total = kwargs.get("retry_total", 3)  # type: int
         self.max_retries = self.retry_total  # type: int
-        self.retry_mode = kwargs.get("retry_mode", RetryMode.EXPONENTIAL)
+        self.retry_mode = RetryMode(kwargs.get("retry_mode", 'exponential'))
```
Comment (annatisch) [inline sdk/eventhub/azure-eventhub/azure/eventhub/_configuration.py:21]: Will this work if someone _is_ passing in the Core RetryMode enum?
e.g. `RetryMode(RetryMode.Fixed))`

File: sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_client.py [hunk 2]
```diff
@@ -60,8 +60,8 @@ class ServiceBusClient(object):
     :keyword float retry_backoff_factor: Delta back-off internal in the unit of second between retries.
      Default value is 0.8.
     :keyword float retry_backoff_max: Maximum back-off interval in the unit of second. Default value is 120.
-    :keyword retry_mode: Fixed or exponential delay between attempts, default is exponential.
-    :paramtype retry_mode: ~azure.servicebus.RetryMode
+    :keyword retry_mode: 'fixed' or 'exponential' delay between attempts, default is 'exponential'.
```
Comment (annatisch) [inline sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_client.py:63]: I would rephrase this to be more explicit about the supported values:
`The delay behaviour between retry attempts. Supported values are 'fixed' or 'exponential', where 'exponential' is the default.`

File: sdk/eventhub/azure-eventhub/azure/eventhub/_configuration.py (hunk 1 above)
Comment (swathipil) [reply sdk/eventhub/azure-eventhub/azure/eventhub/_configuration.py:21]: yes it will - just tested!

Comment (swathipil) [reply sdk/eventhub/azure-eventhub/azure/eventhub/_configuration.py:21]: added a test to validate this

Comment (annatisch) [reply sdk/eventhub/azure-eventhub/azure/eventhub/_configuration.py:21]: Awesome thanks!

**Review by annatisch — APPROVED** [review]
